sceptre/stack.py

```diff
@@ -514,19 +514,65 @@ def validate_template(self):
         )
         return response
 
+    def stack_exists(self):
+        """
+        Returns whether the stack already exists.
+
+        :returns: Whether the given stack exists.
+        :rtype: bool
+        :raises: botocore.exceptions.ClientError
+        """
+        try:
+            response = self.describe()
+            if len(response["Stacks"]) != 1:
+                return False
+
+            # When you run CreateChangeSet on a a stack that does not exist,
+            # CloudFormation will create a stack and set it's status
+            # REVIEW_IN_PROGRESS. However this stack is cannot be manipulated
+            # by "update" commands. Under this circumstances, we treat like
+            # this stack does not exist and call CreateChangeSet will
+            # ChangeSetType set to CREATE and not UPDATE.
+            stack = response["Stacks"][0]
+            return stack["StackStatus"] != "REVIEW_IN_PROGRESS"
+
+        except botocore.exceptions.ClientError as e:
+            # If a stack does not exist, describe_stacks will throw an
+            # exception. Unfortunately we don't have a better way than parsing
+            # the exception msg to understand the nature of this exception.
+            msg = str(e)
+
+            if (
+                    "Stack with id {0} does not exist".format(
+                        self.external_name)
+                    in msg
+            ):
+                self.logger.debug(
+                    "%s - Stack with id %s does not exist".format(
+                    self.name, self.external_name))
+                return False
+            else:
+                # We don't know anything about this exception. Don't handle
+                self.logger.debug("%s - Unable to get stack details.",
+                        self.name, exc_info=e)
+                raise e
+
     def create_change_set(self, change_set_name):
         """
         Creates a change set with the name ``change_set_name``.
 
         :param change_set_name: The name of the change set.
         :type change_set_name: str
+        :raises: botocore.exceptions.ClientError
         """
+        change_set_type = "UPDATE" if self.stack_exists() else "CREATE"
```

Comment:
Could I suggest you mimic the pattern `launch()` - You could go ahead, assume UPDATE change_set_type and make the API call to create_change_set. Boto will throw an error if stack doesn't exist... catch that and retry the API call with CREATE change_set_type. This will also save an API call with an UPDATE change_set succeeds.

Reply:
I'll give that a shot. I mimicked the pattern used in awscli's [`deploy`](http://docs.aws.amazon.com/cli/latest/reference/cloudformation/deploy/index.html) command which first makes this call to decide how to proceed, borrowing that pattern blindly assuming that Amazon had thought it through, but without much of my own justification (asking forgiveness rather than permission is typically nicer, I agree).